test/jdk/sun/net/www/http/HttpClient/MultiThreadTest.java

```diff
@@ -138,62 +159,96 @@ public static void main(String args[]) throws Exception {
         ss.bind(new InetSocketAddress(loopback, 0));
         Server svr = new Server(ss);
         svr.start();
+        var latch = new CountDownLatch(threads);
 
-        Object lock = MultiThreadTest.getLock();
-        synchronized (lock) {
-            for (int i=0; i<threads; i++) {
-                MultiThreadTest t = new MultiThreadTest(svr.getAuthority(), requests);
-                t.start ();
+        MAIN_START = System.nanoTime();
+        try {
+            Object lock = MultiThreadTest.getLock();
+            List<MultiThreadTest> tests = new ArrayList<>();
+            for (int i = 0; i < threads; i++) {
+                MultiThreadTest t = new MultiThreadTest(svr.getAuthority(), requests, latch);
+                tests.add(t);
+                t.start();
             }
-            try {
-                lock.wait();
-            } catch (InterruptedException e) {}
-        }
 
-        // shutdown server - we're done.
-        svr.shutdown();
-
-        int cnt = svr.connectionCount();
-        MultiThreadTest.debug("Connections = " + cnt);
-        int reqs = Worker.getRequests ();
-        MultiThreadTest.debug("Requests = " + reqs);
-        System.out.println ("Connection count = " + cnt + " Request count = " + reqs);
-
-        // We may have received traffic from something else than
-        // our client. We should only count those workers for which
-        // the expected header has been found.
-        int validConnections = 0;
-        for (Worker w : svr.workers()) {
-            if (w.headerFound) validConnections++;
-        }
+            latch.await();
+            long end = System.nanoTime();
+            DONE.compareAndSet(false, true);
+            for (var test : tests) test.join();
+
+            MultiThreadTest.debug("DONE at " + at(end) + "ms");
+
+            // shutdown server - we're done.
+            svr.shutdown();
+
+            int cnt = svr.connectionCount();
+            MultiThreadTest.debug("Connections = " + cnt);
+            int reqs = Worker.getRequests();
+            MultiThreadTest.debug("Requests = " + reqs);
+            System.out.println("Connection count = " + cnt + " Request count = " + reqs);
+
+            // We may have received traffic from something else than
+            // our client. We should only count those workers for which
+            // the expected header has been found.
+            int validConnections = 0;
```

Comment:
Could you also add a comment like:

// We detect worker threads that may have timed out, so we don't include them in
// the count to compare with the number of connections.

Otherwise, LGTM

Reply:
Done!